sdk/core/azure-core/tests/test_pipeline.py

```diff
@@ -204,6 +204,13 @@ def test_format_url_double_query():
     assert formatted == "https://bing.com/path/subpath?query=testvalue&x=2ndvalue&a=X&c=Y"
 
 
+def test_format_url_braces_with_dot():
```

Comment:
Can we add a test for the single bracket scenario that was outlined as well please?